### DISCUSSION TITLE
Game.ini & Engine.ini overwritten

### DISCUSSION THREAD

**Discussion by cooljam105** [wolveix/satisfactory-server#2] [Q&A] [answered]
Just want to say this is awesome. I have been running this on my kubernetes cluster for a bit over 24 hours now and it seems to be stable.

One question I have is it appears that the Engine.ini and Game.ini files are being overwritten every time i start up the container, effectively erasing any changes I had made previously. When I set it to read-only the init.sh script fails because it can not touch those files. 

any ideas on how I can stop that from happening?

**Comment by wolveix** [comment, marked as answer]
Hey! I'm glad you're enjoying using it :D 

Yeah, the files had to be overwritten on reset as Satisfactory removes a few of the options that we need.

I've just modified where the files are overwritten from, though. Now, instead of being filled from `/root/Engine.ini` and `/root/Game.ini`, they're now filled from `/config/Engine.ini` and `/config/Game.ini`. If those files don't exist, they'll be populated with the default config that the server needs.

So essentially pull the latest image, reboot your container, then add any custom config options to the files in `/config` and they'll apply (and persist) through container restarts :) 

I'm curious though, what customizations are you setting?

**Reply by cooljam105** [reply]
Thanks for the quick update, much appreciated. 

I wonder if it is different because I am running this on Kubernetes? The container basically gets destroyed and recreated on restart. I can see that /config/Game.ini and /config/Engine.ini are being copied over as expected. However it seems like the game is overwriting those files as soon as it starts up. So far the only fix I have seen posted by folks is to make those files read-only so the game can't change them, however as I stated previously doing so breaks the init.sh script.

For the settings, I am just changing what is listed in the [wiki](https://satisfactory.fandom.com/wiki/Multiplayer) , Just to fix the lag issues,

**Reply by wolveix** [reply]
@cooljam105 Ah damn, I hadn't noticed that the game was overwriting the files 🤦‍♂️ The latest version addresses this 😄 The default config overrides that the repo provides (per the wiki's recommendation) will actually take effect now, and won't be overwritten.

**Reply by cooljam105** [reply]
@wolveix Yes that seemed to do the trick. I can now verify that the server is no longer overwriting my config changes. Thanks for fixing this so quickly.

As a side note, this server is working flawlessly, I have been running it for about 10 days now and I really haven't had any issues other than the Early Access quirks. Great work on this!!

**Reply by wolveix** [reply]
@cooljam105 I appreciate the kind words! Likewise, it's been running flawlessly for us too over the last 2 weeks or so :) I previously gave it a full baremetal server to run on, but quickly realised that it's nowhere near as intensive as I had previously assumed it to be; so I ended up moving it to Digital Ocean's Kubernetes offering and haven't had any issues there.
